src/waterfall/waterfall.cpp

```diff
@@ -396,6 +397,14 @@ void Waterfall::hoverMoveEvent(QHoverEvent *event)
 void Waterfall::hoverLeaveEvent(QHoverEvent *event)
 {
     Q_UNUSED(event)
-    emit mouseLeave();
     _mouseDepth = -1;
```

Comment:
```suggestion
    // TODO why, is this necessary?
    _mouseDepth = -1;
```

Reply:
MouseDepth can be used to get the depth where the mouse is pointing.

Reply:
The doxygen documentation:
```
     /**
     * @brief Get depth from mouse position
     *
     * @return float
      */
    float mouseDepth() {return _mouseDepth;}
    Q_PROPERTY(float mouseDepth READ mouseDepth NOTIFY mouseDepthChanged)
```

Reply:
I couldn't find usages of `mouseDepth`, but i found of `mouseColumnDepth`.
Are they different things?
And are both necessary?

Reply:
The doxygen documentation says:
```cpp
/**
 * @brief Get depth from mouse position
 *
 * @return float
 */
float mouseDepth() {return _mouseDepth;}
Q_PROPERTY(float mouseDepth READ mouseDepth NOTIFY mouseDepthChanged)

/**
 * @brief Get signal deepth from mouse position column
 *
 * @return float
 */
float mouseColumnDepth() {return _mouseColumnDepth;}
Q_PROPERTY(float mouseColumnDepth READ mouseColumnDepth NOTIFY mouseColumnDepthChanged)
```

The first one is the mouse position depth, the second one is the mouse column depth

Reply:
@patrickelectric I mean why is it necessary to set to -1?

Reply:
If mouse is not in the application, it sets the depth to an invalid value.

Reply:
So if mouse enters application, do we need to reinitialize to a valid value?

Reply:
This is done automatically by `hoverMoveEvent`

Reply:
```suggestion
    // The mouse is not inside the application, so set the depth under the mouse to an invalid value
    _mouseDepth = -1;
```